pkg/gardenlet/controller/networkpolicy/reconciler.go

```diff
@@ -33,35 +34,21 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-// namespaceReconciler implements the reconcile.Reconcile interface for namespace reconciliation.
-type namespaceReconciler struct {
-	seedClient             client.Client
-	seedName               string
-	shootNamespaceSelector labels.Selector
-	resolver               hostnameresolver.HostResolver
-}
-
-// newNamespaceReconciler returns the new namespace reconciler.
-func newNamespaceReconciler(
-	seedClient client.Client,
-	seedName string,
-	shootNamespaceSelector labels.Selector,
-	resolver hostnameresolver.HostResolver,
-) reconcile.Reconciler {
-	return &namespaceReconciler{
-		seedClient:             seedClient,
-		seedName:               seedName,
-		shootNamespaceSelector: shootNamespaceSelector,
-		resolver:               resolver,
-	}
+// Reconciler implements the reconcile.Reconcile interface for namespace reconciliation.
+type Reconciler struct {
+	Config                 config.SeedAPIServerNetworkPolicyControllerConfiguration
+	SeedClient             client.Client
+	SeedName               string
+	ShootNamespaceSelector labels.Selector
```

Comment:
Why do we need to make this configurable? I don't see a use case when we would overwrite this?

Reply:
ok, `shootNamespaceSelector` is private now

Reply:
It is not needed to be a struct attribute at all, or?